src/jj_terms.jl

```diff
@@ -15,19 +14,26 @@ function couple_terms(J::Vector{N}, j₀::N=zero(N)) where {I<:Integer,R<:Ration
     ts
 end
 
+couple_terms(J1::Real, J2::Real) =
+    couple_terms(convert(HalfInteger, J1), convert(HalfInteger, J2))
+couple_terms(J::Vector{T}, j₀::Real=zero(T)) where {T <: Real} =
+    couple_terms(convert.(HalfInteger, J), convert(HalfInteger, j₀))
+
 function jj_terms(orb::RelativisticOrbital, w::Int=one(Int))
-    @unpack ℓ,j = orb
+    @unpack ℓ, j = orb
+
+    w <= 2j+1 || throw(ArgumentError("w=$w too large for $orb orbital"))
 
-    2w ≥ 2j+1 && (w = 2j+1-w)
-    w == 0 && return [zero(Rational{Int})]
+    2w ≥ 2j+1 && (w = convert(Int, 2j) + 1 - w)
+    w == 0 && return [zero(HalfInteger)]
     w == 1 && return [j]
 
     # Forms full Cartesian product of all mⱼ, not necessarily the most
     # performant method.
     mⱼs = filter(allunique, collect(allchoices([-j:j for i = 1:w])))
-    MJs = map(sum, mⱼs)
+    MJs = map(x -> reduce(+, x), mⱼs) # TODO: make map(sum, mⱼs) work
```

Comment:
For some reason, e.g. `sum(::Vector{HalfInteger})` promotes to `Float64`, which I couldn't figure out. E.g.:

```
julia> sum([hi"1/2", hi"1/2"])
1.0
```

Reply:
I see the issue with Julia 1.1.0-DEV.839 (b55b85cd99) and 1.2.0-DEV.31 (e6938a052a) with the WignerSymbols one too:

```
julia> using WignerSymbols

julia> x,y = WignerSymbols.HalfInteger(1), WignerSymbols.HalfInteger(2)
(WignerSymbols.HalfInteger(1), WignerSymbols.HalfInteger(2))

julia> sum([x,y]) |> typeof
Float64
```